### DISCUSSION TITLE
Reduce app size

### DISCUSSION THREAD

**Discussion by baseplate-admin** [flet-dev/flet#2319] [Ideas]
Hi, It seems that files generated by flet are large in nature. I was wondering if it was possible to reduce size of the binary

Possible options :
* Tree Shake Codes ( maybe do conditional installs of packages, if i dont use a flutter package, don't install that )
* Reduce python venv size  ( by using something like [`dead`](https://github.com/asottile/dead), [`vulture`](https://pypi.org/project/vulture/)  can help with this ) )

---

Cross posted in : https://github.com/orgs/kivy/discussions/22#discussion-6032762

**Comment by ndonkoHenri** [comment]
Can you try the suggestions here: https://docs.flutter.dev/perf/app-size

**Reply by baseplate-admin** [reply]
Yep. Flet already optimizes according to the docs.

I was wondering more on the python side of things

**Reply by FeodorFitsner** [reply]
I assume we are talking about `.apk`?

Python library itself is 5.2 MB - I'm not sure it could be easily reduced in size and IMHO it's not a lot.

<img width="660" alt="image" src="https://github.com/flet-dev/flet/assets/5041459/050f7f66-9ad3-451e-a545-8d47f6bcce06">

`libpythonbundle.so` on the other hand is, in reality, a .zip and I think it could be reduced in size:

<img width="831" alt="image" src="https://github.com/flet-dev/flet/assets/5041459/de437eff-ab9d-4ff1-8001-167c7d78ec50">

`dead` and `vulture` are not exact tools that help, we'd need a tree-shaking algorithm for Python itself, kind of what PyInstaller has.

**Reply by baseplate-admin** [reply]
> libpythonbundle.so on the other hand is, in reality, a .zip and I think it could be reduced in size:

Correct. This is what i wanr to reduce. The size is due to 

1. `python-for-android` bundles the entire venv for the final `.zip` bundle. In an Ideal scenario, this shouldn’t be the case (you only ship the bare minimum for the app to function)


> we'd need a tree-shaking algorithm for Python itself, kind of what PyInstaller has.

I would suggest this to be implemented in `python-for-android` repo itself.

 @julian-o, @misl6, how would you guys feel about integrating it into `python-for-android` toolchain

**Reply by Julian-O** [reply]
I have no comment as I have no technical understanding of the challenges.. However, I would suggest rhis discussion belongs where they python-for-android developers can see it, rather than clogging the Flet channels.

**Reply by baseplate-admin** [reply]
>  I would suggest rhis discussion belongs where they python-for-android developers can see it, rather than clogging the Flet channels.

I second this. @FeodorFitsner, would you kindly discuss this (and possibly help the core devs) in `python-for-android`? This improvement will benefit the entire `python-android` ecosystem

**Comment by baseplate-admin** [comment]
Hi is there any progress on this?

**Comment by zcattacz** [comment]
how about upx the `.so`s
